io.openems.backend.metadata.user_based/src/io/openems/backend/metadata/user_based/UserBased.java

```diff
@@ -0,0 +1,207 @@
+package io.openems.backend.metadata.user_based;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import io.openems.backend.common.component.AbstractOpenemsBackendComponent;
+import io.openems.common.access_control.AccessControl;
+import io.openems.common.access_control.RoleId;
+import io.openems.common.channel.Level;
+import io.openems.common.utils.FileUtils;
+import io.openems.backend.metadata.api.BackendUser;
+import io.openems.backend.metadata.api.Edge;
+import io.openems.backend.metadata.api.Edge.State;
+import io.openems.backend.metadata.api.IntersectChannels;
+import io.openems.backend.metadata.api.Metadata;
+import io.openems.common.exceptions.OpenemsError;
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.common.session.Role;
+import io.openems.common.types.ChannelAddress;
+import io.openems.common.types.EdgeConfig;
+import io.openems.common.utils.JsonKeys;
+import io.openems.common.utils.JsonUtils;
+import org.osgi.service.component.annotations.*;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * TODO Fill the comment as soon as the logic works
+ */
+@Designate(ocd = Config.class, factory = false)
+@Component(name = "Metadata.UserBased", configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class UserBased extends AbstractOpenemsBackendComponent implements Metadata, IntersectChannels {
+
+    private final Logger log = LoggerFactory.getLogger(UserBased.class);
+    private final Map<String, BackendUser> sessionIdUserMap = new HashMap<>();
+    private final Map<String, BackendUser> userMap = new HashMap<>();
+    private final Map<BackendUser, String> userPasswordMapping = new HashMap<>();
+    private final Map<String, Edge> edges = new HashMap<>();
+    private String path = "";
+    private AtomicInteger sessionId = new AtomicInteger(0);
+
+    public UserBased() {
+        super("Metadata.UserBased");
+    }
+
+    @Activate
+    void activate(Config config) {
+        log.info("Activate [path=" + config.path() + "]");
+        this.path = config.path();
+
+        // Read the data async
+        CompletableFuture.runAsync(this::refreshData);
```

Comment:
The downside of reading the file async is, that you will never get notified about an Exception.